### ISSUE TITLE
Difficultés avec la modélisation du domaine (LOG210)

### ISSUE THREAD

**Issue by fuhrmanator** [fuhrmanator/log210-ndc-quarto#114]
Larman mentionne plusieurs classes à ne pas mettre dans les MDD:

- artefact logiciel, p. ex. BDVentes (confusion avec la catégorie de classe conceptuelle Systèmes Externes). 
- classes logicielles (avec méthodes)

La clientèle de LOG210 à l'ÉTS spécifie des associations entre classes conceptuelles parfois avec des attributs dans les classes qui ressemblent une classe logicielle. P. ex. la relation entre Caisse et Vente aura une classe Vente avec un attribut `caisse : Caisse` et une classe Caisse avec un attribut `ventes: Array<Vente>`. Cette erreur est reliée aux directives de Larman dans 9.14 Associations:

> During domain modeling, an association is _not_ a statement about data flows, database foreign key relationships,
instance variables, or object connections in a software solution; it is a statement that a relationship is
meaningful in a purely conceptual perspective in the real domain.

> That said, many of these relationships _will_ be implemented in software as paths of navigation and visibility (both
in the Design Model and Data Model). But the domain model is not a data model; associations are added to
highlight our rough understanding of noteworthy relationships, not to document object or data structures.

Il serait bien qu'on résume ces problèmes dans le chapitre 4 du manuel quelque part.

**Comment by Olon11** [comment]
Si ça ne presse pas trop, je peux le faire. J'aurais du temps en septembre.

**Comment by Olon11** [comment]
Fait cocasse :

En LOG210, les étudiants veulent ajouter des attributs pour représenter les associations dans le MDD pour faire comme en LOG121.

En GTI311, les étudiants ne veulent pas ajouter des attributs dans les diagrammes de classes pour représenter les associations pour faire comme en LOG210 🙃.

**Comment by fuhrmanator** [comment]
Aha - la confusion semble être entre un modèle du domaine (problème) et d'un logiciel (solution). Je sens un exercice à ce propos. Qu'en penses-tu?

**Comment by Olon11** [comment]
Oui, il y a clairement une confusion entre le domain de l'analyse et de la conception. Plusieurs étudiants apellent tous les diagrammes de séquence des DSS après avoir fait LOG210.

Je crois qu'un exercice serait pertinent. Une idée : on donne un MDD qui contient des éléments d'un diagramme de classes (ou l'inverse). Les étudiants doivent identifier les erreurs et les corriger.
